### PR TITLE
workflow: resolve readability comments for canary feature.

### PR DESCRIPTION
**Pull request by wangyipei01**

### CODE REVIEW

File: go/vt/workflow/resharding/parallel_runner.go [hunk 1]
```diff
@@ -63,13 +62,13 @@ type ParallelRunner struct {
 // NewParallelRunner returns a new ParallelRunner.
 func NewParallelRunner(ctx context.Context, rootUINode *workflow.Node, cp *CheckpointWriter, tasks []*workflowpb.Task, executeFunc func(context.Context, *workflowpb.Task) error, concurrencyLevel level, enableApprovals bool) *ParallelRunner {
 	if len(tasks) < 1 {
-		log.Fatal(errors.New("BUG: No tasks passed into ParallelRunner"))
+		log.Fatalf("BUG: No tasks passed into ParallelRunner")
```
Comment (michael-berlin) [inline go/vt/workflow/resharding/parallel_runner.go]: Please use `log.Fatal` (without f) if the parameter should not be interpreted as format string. This mostly important if you log the content of a variable e.g. `log.Fatal(err)`. But to be consistent, let's do this for string literals as well.

File: go/vt/workflow/resharding/parallel_runner.go [hunk 2]
```diff
@@ -225,10 +223,10 @@ func (p *ParallelRunner) triggerRetry(taskID string) error {
 	// Disable the retry action and synchronize for retrying the job.
 	node, err := p.rootUINode.GetChildByPath(taskID)
 	if err != nil {
-		panic(fmt.Sprintf("BUG: node on child path %v not found", taskID))
+		log.Fatalf("BUG: node on child path %v not found", taskID)
 	}
 	if len(node.Actions) == 0 {
-		panic(fmt.Sprintf("BUG: node actions should not be empty"))
+		log.Fatalf("BUG: node actions should not be empty")
```
Comment (michael-berlin) [inline go/vt/workflow/resharding/parallel_runner.go]: Same comment as above.

Comment (wangyipei01) [inline go/vt/workflow/resharding/parallel_runner.go]: Done.

File: go/vt/workflow/resharding/parallel_runner.go (hunk 1 above)
Comment (wangyipei01) [inline go/vt/workflow/resharding/parallel_runner.go]: Done.

**Comment by wangyipei01** [comment]
LTGM

**Pull request merged**